### PR TITLE
Refactor code to rely on CreateDiagnostic helper

### PR DESCRIPTION
**Pull request by Evangelink**

### CODE REVIEW

**Comment by mavasani** [comment]
@Evangelink there are test failures

File: src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Security/DataSetDataTableInWebSerializableObjectGraphTests.cs [hunk 1]
```diff
@@ -31,7 +30,7 @@ public string MyWebMethod(DataTable dataTable)
     }
 }
 ",
-                GetCSharpResultAt(10, 31, "DataTable", "DataTable"));
+                GetCSharpResultAt(10, 41, "DataTable", "DataTable"));
```
Comment (Evangelink) [inline src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Security/DataSetDataTableInWebSerializableObjectGraphTests.cs:33]: With the old technique, the first column of the location was the data-type, now we "correctly" report on the parameter.

Comment (mavasani) [reply src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Security/DataSetDataTableInWebSerializableObjectGraphTests.cs:33]: Tagging @dotpaul as this is a security rule.

File: src/Utilities/Compiler/Extensions/DiagnosticExtensions.cs [hunk 2]
```diff
@@ -144,6 +154,29 @@ internal static class DiagnosticExtensions
                      messageArgs: args);
         }
 
+        public static Diagnostic CreateDiagnostic(
+            this SyntaxReference syntaxReference,
```
Comment (mavasani) [inline src/Utilities/Compiler/Extensions/DiagnosticExtensions.cs:158]: I don't really like all these SyntaxReference extension methods, seem to be very specific to a set of internal analyzers in Roslyn.Diagnostic.Analyzers. Maybe it is better to move all of them to just that project inside a separate helper type?

Comment (Evangelink) [reply src/Utilities/Compiler/Extensions/DiagnosticExtensions.cs:158]: I am fine with this.

File: src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Security/DataSetDataTableInWebSerializableObjectGraphTests.cs (hunk 1 above)
Comment (dotpaul) [reply src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Security/DataSetDataTableInWebSerializableObjectGraphTests.cs:33]: In this case, it's the type that's the issue, so I think I prefer the older technique's location.

Also, could you make all the security rule changes in the 2.9.x branch please?

Comment (Evangelink) [reply src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Security/DataSetDataTableInWebSerializableObjectGraphTests.cs:33]: Sure for the 2.9.x and regarding the location I wasn't sure as it was originally reporting on the full parameter (type + var name), I'll change it.

**Review by mavasani — APPROVED** [review]

File: src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Security/DataSetDataTableInWebSerializableObjectGraphTests.cs (hunk 1 above)
Comment (dotpaul) [reply src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Security/DataSetDataTableInWebSerializableObjectGraphTests.cs]: Thanks @Evangelink !

**Pull request merged**